app/Http/Controllers/Secretariat/SemesterController.php

```diff
@@ -32,10 +32,10 @@ public function showStatusUpdate()
         if ($user->getStatusIn(Semester::previous()->id) == SemesterStatus::DEACTIVATED) {
             abort(403);
         }
-        if (!self::isStatementAvailable()) {
+        if (!self::isStatementAvailable($user)) {
             abort(403);
```

Comment:
return redirect home with a session message that the status has been set already